api/src/api/admin.js

```diff
@@ -166,17 +166,141 @@ router.post(
   }),
 );
 
+router.put(
+  '/categories/:id',
+  errorWrap(async (req, res) => {
+    const { id } = req.params;
+    const updatedCategory = await Category.findByIdAndUpdate(id, req.body, {
+      new: true,
+      runValidators: true,
+    });
+
+    const updatedResources = await Resource.updateMany(
+      { category: req.body.currentName },
+      { $set: { 'category.$': req.body.newName } },
+    );
+
+    res.json({
+      code: 200,
+      message: `Successfully updated category ${id}`,
+      success: true,
+      result: {
+        updatedCategory,
+        updatedResources,
+      },
+    });
+  }),
+);
+
 // Delete a category
 router.delete(
   '/categories/:id',
   errorWrap(async (req, res) => {
     const { id } = req.params;
     await Category.findByIdAndDelete(id);
+
+    const updatedResources = await Resource.updateMany(
+      { category: req.body.currentName },
+      { $pull: { category: req.body.currentName } },
+    );
+
     res.json({
       code: 200,
       message: `Successfully deleted category ${id}`,
       success: true,
-      result: null,
+      result: updatedResources,
+    });
+  }),
+);
+
+// Create a new subcategory
+router.post(
+  '/subcategories/:id',
+  errorWrap(async (req, res) => {
+    const { id } = req.params;
+    const updatedCategory = await Category.findByIdAndUpdate(
+      id,
+      { $push: { subcategories: req.body.name } },
+      {
+        new: true,
+        runValidators: true,
+      },
+    );
+    res.status(201).json({
+      code: 201,
+      message: `Successfully created new category ${req.body.name}`,
+      success: true,
+      result: updatedCategory,
+    });
+  }),
+);
+
+// renaming subcategory
+router.put(
+  '/subcategories/:id',
+  errorWrap(async (req, res) => {
+    const { id } = req.params;
+
+    const updatedCategory = await Category.findOneAndUpdate(
+      {
+        _id: req.params.id,
+        subcategories: req.body.currentName,
+      },
+      {
+        $set: { 'subcategories.$': req.body.newName },
+      },
+      {
+        new: true,
+        runValidators: true,
+      },
+    );
+
+    const updatedResources = await Resource.updateMany(
```

Comment:
Similarly with the renaming subcategory comment I made, I think we can just get rid of this `updatedResources` variable